### DISCUSSION TITLE
Allow creating custom cli commands

### DISCUSSION THREAD

**Discussion by hardfire** [blitz-js/blitz#1060] [Ideas]
I very often see myself creating the first user (or other db objects) using the `blitz console`. It would be pretty cool if we could create custom cli commands to do activities that are repeated often.  I see this pattern being used extensively in [laravel](https://laravel.com/docs/7.x/artisan).

PS: apologies if this has already been discussed before, couldn't find that discussion.

**Comment by flybayer** [comment]
Hey! Yeah we could add a `blitz task` command that make it easy to do this stuff.

Currently the best way is to write Typescript scripts that you run via ts-node via package.json scripts. Check out [this comment](https://github.com/blitz-js/blitz/issues/952#issuecomment-682268003) on how to make it work with absolute paths.

**Reply by davidharting** [reply]
Another great thing about blitz task (or similar) would be that we could start other processes without building out a separate TypeScript tool chain. 

For instance, I would love to be able to start BullMQ workers using Blitz itself.

**Comment by MrLeebo** [comment]
I also use blitz console for this kind of stuff, and would create a "scripts" folder with service functions that are easy to import and invoke.

```ts
// blitz console
await require('scripts/devUserCreator').create()
```

It's a little more clunky than a pure CLI solution, but you get type safety on the function arguments.

**Reply by davidharting** [reply]
This is a good way to handle it! I would imagine you can write tests for your scripts that are executed by the out-of-the-box jest setup too which is something else I want. 

I would be happy as well if I could just pass a script to blitz console instead of having a new command.

**Comment by prisis** [comment]
For the same needs i created (need some time to finish the recipe) https://github.com/anolilab/blitzjs-cli-recipe/tree/feature/cli and https://github.com/anolilab/cerebro/tree/alpha

Im using a custom webpack config to search the commands like blitz for pages (preview https://discord.com/channels/802917734999523368/843912213399601192/875063361115656212)
